src/metrics/include/metrics/metrics.hpp

```diff
@@ -92,13 +140,23 @@ class RegistryT final {
                                       std::string opName,
                                       ActorId actorId,
                                       std::optional<genny::PhaseNumber> phase = std::nullopt) {
+        std::string name;
+        if (_format.use_grpc()) {
+            name = createName(actorName, opName, phase);
+            _collectors.try_emplace(name, name, _path_prefix);
+        }
         auto& opsByType = this->_ops[actorName];
         auto& opsByThread = opsByType[opName];
-        auto opIt =
-            opsByThread
-                .try_emplace(
-                    actorId, std::move(actorName), *this, std::move(opName), std::move(phase))
-                .first;
+        auto opIt = opsByThread
+                        .try_emplace(actorId,
+                                     actorId,
+                                     std::move(actorName),
```

Comment:
er..I think I'm confused about how this works/why this changed - we didn't change anything under opsByThread so not sure why we now need to pass actorId twice (or really why any change to this portion was necessary)? I'm like 100% reading this wrong though I'd guess.

Reply:
We added actorId to the constructor for OperationImpl. So now the first actorId is used as the key for the map and the second actorId is passed into the constructor. We needed to pipe in the actorId so that it can be piped into the id field in poplar's EventMetrics.